edx_repo_tools/codemods/django3/add_new_django32_settings.py

```diff
@@ -0,0 +1,80 @@
+"""
+Codemod to add new settings in the repo settings file.
+"""
+import json
+import re
+import os
+import click
+from copy import deepcopy
+
+
+NEW_HASHING_ALGORITHM = "sha1"
+NEW_AUTO_FIELD = "django.db.models.AutoField"
+New_PROCESSOR = "django.template.context_processors.request"
+
+def update_hash_algorithm(file_data, new_algorithm=NEW_HASHING_ALGORITHM):
+    """
+    Update the HASHING_ALGORITHM in the file data and append if not found already.
+    """
+    algorithm_key = "DEFAULT_HASHING_ALGORITHM"
+    # if found the pattern, Update the algorithm value
+    if re.search(algorithm_key, file_data):
+        file_data = re.sub(
+            algorithm_key+" = .*\\n",
```

Comment:
Tested the script again with the given string. Script matches the pattern correctly and also doesn't add another if one is already present. 
I've wrote the pattern according to the string we get when reading data from file so that's why it isn't matching the given string directly.

Reply:
could you try testing the pattern here: https://regex101.com/r/1XSjcG/1 ?